docs/plugins/data_persistence.md

```diff
@@ -0,0 +1,232 @@
+    2. Global query state is persistent across the entire application, values will persist when we refresh the page, or when we navigate across visualize, discover, timeline or dashboard page. For example, if we set time range to last 24 hours, and refresh intervals to every 30 min, the same time range and refresh intervals will be applied if we navigate to any of the other pages.
+    3. Params:
+       1. global filters (Select `pin filter` to make the filters global)
+
+       ![img](./img/global_filter.png)
+
+       2. refresh intervals
+
+       ![img](./img/refresh_interval.png)
+
+       3. time range
+
+       ![img](./img/time_range.png)
+
+# URL breakdown & example
+
+![img](./img/URL_example.png)
+# Global state persistence
+
+1. In plugin.ts, during plugin setup, call `createOsdUrlTracker()`, listen to history changes and global state changes, then update the nav link URL. This also returns function such as `onMountApp()`, `onUnmountedApp()`
+    ```ts
+            const {
+        appMounted,
+        appUnMounted,
+        ...
+        } = createOsdUrlTracker({
+        baseUrl: core.http.basePath.prepend('/app/visualize'),
+        defaultSubUrl: '#/',
+        storageKey: `lastUrl:${core.http.basePath.get()}:visualize`,
+        navLinkUpdater$: this.appStateUpdater,
+        stateParams: [
+        {
+            osdUrlKey: '_g',
+            stateUpdate$: data.query.state$.pipe(
+            filter(
+                ({ changes }) => !!(changes.globalFilters || changes.time || changes.refreshInterval)
+            ),
+            map(({ state }) => ({
+                ...state,
+                filters: state.filters?.filter(opensearchFilters.isFilterPinned),
+            }))
+            ),
+        },
+        ],
+        ....
+    ```
+
+    * When we enter the app and app is mounted, it initialize nav link by getting previously stored URL from storage instance: const storedUrl = storageInstance.getItem(storageKey). (Storage instance is a browser wide session storage instance.) Then it unsubscribes to global state$ and subscribes to URL$. The current app actively listens to history location changes. If there are changes, set the updated URL as the active URL
+
+    ```ts
+      function onMountApp() {
+      unsubscribe();
+      ...
+      // track current hash when within app
+            unsubscribeURLHistory = historyInstance.listen((location) => {
+                  ...
+                  setActiveUrl(location.hash.substr(1));
+               }
+            });
+         }
+    ```
+
+    * When we are leaving the app and app is unmounted, unsubscribe URL$ and subscribe to global state$. If the global states are changed in another app, the global state listener will still get triggered in this app even though it is unmounted, it will set the updated URL in storage instance, so next time when we enter the app, it gets the URL from the storage instance thus the global state will persist.
```

Comment:
```suggestion
    * When we are leaving the app and app is unmounted, unsubscribe` URL$` and subscribe to global `state$`. If the global states are changed in another app, the global state listener will still get triggered in this app even though it is unmounted, it will set the updated URL in storage instance, so next time when we enter the app, it gets the URL from the storage instance thus the global state will persist.
```